### ISSUE TITLE
Malformed paths in .js.map

### ISSUE THREAD

**Issue by spamguy** [gopherjs/gopherjs#340]
I primarily develop in NodeJS. My Go code lives in GOPATH, but my NodeJS code and generated GopherJS code lives elsewhere. As a result, the generated .js.map file is full of references to invalid paths:

```
{"version":3,"file":"judge.js","sources":["/github.com/gopherjs/gopherjs/js/js.go","/github.com/gopherjs/gopherjs/compiler/natives/runtime/runtime.go"...]}
```

Were these preceded by $GOPATH they would probably be fine.

This broken .js.map file makes debugging in node-inspector impossible. Debugging aborts every time I pass over a package reference. If I delete the source map reference I can debug fully.

Suggestions:
* Provide an option to disable .js.map generation.
* Prefix file paths with $GOPATH if appropriate. [This guy](https://github.com/dennisfrancis/dygraph-gopherjs-demo/blob/master/graph.js.map) has them, so I don't know how I missed out.

**Comment by slimsag** [comment]
> Provide an option to disable .js.map generation.

Highly agree! :+1: 

> Prefix file paths with $GOPATH if appropriate.

I don't think this is a good idea. Having the filepaths rooted at `$GOPATH/src` means that you only need to serve the `$GOPATH/src` directory at `/` yourself -- which is highly convenient IMO. Instead of this, I would propose adding an option to set a path prefix -- which you yourself could set to `$GOPATH/src`.

There is also the problem that this option wouldn't work for source files from `$GOROOT` i.e. Go stdlib source code, so a separate filepath prefix option may be needed for those files as well.

(note: I'm just another random users of GopherJS)

**Comment by spamguy** [comment]
> I don't think this is a good idea. Having the filepaths rooted at $GOPATH/src means that you only need to serve the $GOPATH/src directory at / yourself -- which is highly convenient IMO. Instead of this, I would propose adding an option to set a path prefix -- which you yourself could set to $GOPATH/src.

Yeah, this is a better idea than rooting at $GOPATH. Guys, listen to him.

**Comment by neelance** [comment]
The problem here is that there is `$GOROOT` and `$GOPATH` and the latter may even have multiple directories. So there are two or more directories where your Go source may live. The paths in the source map are not file paths, but URLs. A fixed prefix would only solve your problem for one of the paths. So the solution that I prefer is to make the server of the source files overlay those directories on top of each other, just like the `godoc` server does. This behavior is implemented in `gopherjs serve`, so if you use that then source maps would work correctly.

But it seems like you use case involves Node.js. In that case, full file paths may actually be better, also for stack traces. A flag might be useful in this case and `gopherjs run` and `gopherjs test` could use it implicitly.
